src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs

```diff
@@ -246,6 +246,10 @@ private void StartCore()
 
         [DynamicDependency(nameof(StartCallback))]
         [DynamicDependency(nameof(ThrowThreadStartException))]
+#if TARGET_OSX || TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+        [DynamicDependency(nameof(CreateAutoreleasePool))]
```

Comment:
Is this dynamic dependency sufficient to keep these mehods around for all cases? If I have a trivial app that does not create any new threads, `StartInternal` should be unused, but these methods are still going to be called for Main thread. Is it a problem?

Reply:
I believe this is sufficient given ti is called from within Mono and is defined in `icall-def-netcore.h`. I truthfully don't know enough about the Linker and Mono to understand if this is sufficient but defer to Linker and Mono experts here - @vitek-karas @agocke @marek-safar @lambdageek

Reply:
Yea, CreateAutoreleasePool and DrainAutoreleasePool may need to go in the linker descriptor. I'm not sure if there's a way to only add it for specific target platforms.  If there are no calls to Thread.StartInternal, nothing would keep it or its dynamic dependencies alive.

On some platforms (webassembly) icalls in `icall-def-netcore.h` may be linked out if nothing in managed code uses them (look for "IcallTableGenerator").  So the IL linker does not treat icalls specially, AFAIK.  Managed methods invoked from inside Mono are also not automatically preserved.

Reply:
Okay. Will add this to the descriptor as suggested.